src/storage-client/src/sink.rs

```diff
@@ -324,9 +361,17 @@ pub async fn build_kafka(
         KafkaConsistencyConfig::Progress { topic } => ensure_kafka_topic(
             &client,
             topic,
-            1,
-            connection.replication_factor,
-            KafkaSinkConnectionRetention::default(),
+            TopicConfig {
+                partition_count: 1,
+                // TODO: introduce and use `PROGRESS TOPIC REPLICATION FACTOR`
+                // on Kafka connections.
+                replication_factor: -1,
```

Comment:
Note this is not actually a change in behavior because `REPLICATION FACTOR` is an unstable option that is forced to -1 in stable mode. But, were that option stable, it seemed like it'd be very wrong for the progress topic to get created with the replication factor of whichever sink happened to win the race to create the progress topic.